### ISSUE TITLE
fp16 does not work with euler_a scheduler

### ISSUE THREAD

**Issue by KingStorm** [huggingface/diffusers#1484]
### Describe the bug

fp16 does not work with euler_a scheduler: it displays nearly black image.

using checkpoint: stable-diffusion-v1.5

### Reproduction

- Input prompt: a boy.
- euler_a scheduler: default parameters.
- load stable-diffusion-v1.5 with fp16 

### System Info

Ubuntu 18, 2080Ti

**Comment by patrickvonplaten** [comment]
Hey @KingStorm,

I think it's the safety checker disabling your image because it detected nsfw, could you maybe copy-paste a reproducible code snippet here?

**Comment by jwkirchenbauer** [comment]
Possibly related, tried to run the MWE from the model card and am getting fully black images too.

```python
import torch
from diffusers import StableDiffusionPipeline, EulerDiscreteScheduler

model_id = "stabilityai/stable-diffusion-2"

# Use the Euler scheduler here instead
scheduler = EulerDiscreteScheduler.from_pretrained(model_id, subfolder="scheduler")
pipe = StableDiffusionPipeline.from_pretrained(model_id, scheduler=scheduler, revision="fp16", torch_dtype=torch.float16)
pipe = pipe.to("cuda")
# pipe.enable_attention_slicing() # seems to not affect this behavior either way

prompt = "a photo of an astronaut riding a horse on mars"

# orig example size, works fine
# image = pipe(prompt, height=768, width=768).images[0]

# widescreen version was what I wanted, and this works fine too
# image = pipe(prompt, height=512, width=1080).images[0]

# but at the "1080p" res, a black image every time (I think)
image = pipe(prompt, height=1080, width=1920).images[0]
    
image.save("astronaut_rides_horse.png")
```

**Comment by patrickvonplaten** [comment]
@jwkirchenbauer,

Thanks a lot for the reproducible code snippet. Since it's a different issue I opened it for #1552 - it would be awesome if you could open new issues for new problems in the future :-) 

This PR: https://github.com/huggingface/diffusers/actions/runs/3621830689/jobs/6105766777 should solve the problem.

**Comment by jwkirchenbauer** [comment]
Sorry, will do next time! 
Wasn't sure whether it was the fp16 or the euler sched or whatnot, but I found this issue in my initial search.
